src/server/auth/server/api_server.go

```diff
@@ -1115,7 +1115,55 @@ func (a *apiServer) Authenticate(ctx context.Context, req *auth.AuthenticateRequ
 		}); err != nil {
 			return nil, err
 		}
+	case req.IdToken != "":
+		// confirm OIDC has been configured and get OIDC prefix
+		oidcSP := a.getOIDCSP()
+		if oidcSP == nil {
+			return nil, errors.Errorf("error authorizing OIDC id token: no OIDC ID provider is configured")
+		}
+
+		// Determine caller's Pachyderm/OIDC user info (email)
+		claims, err := oidcSP.ValidateJWT(req.IdToken)
```

Comment:
I think our OIDC library has a built-in function for ID token validation, though it could be that it doesn't work for this case. If you haven't seen it, though, I think we use it in `src/server/auth/server/oidc.go:374`:
```
var verifier = sp.Provider.Verifier(&oidc.Config{ClientID: conf.ClientID})
// Extract the ID Token from OAuth2 token.
rawIDToken, ok := tok.Extra("id_token").(string)
if !ok {
	return "", "", errors.New("missing id token")
}

// Parse and verify ID Token payload.
idToken, err := verifier.Verify(ctx, rawIDToken)
if err != nil {
	return "", "", errors.Wrapf(err, "could not verify token")
}
```

Would it make sense to use this (or some version of it) here?

Reply:
Yeah this was an oversight on my part. The oidc-go library looks like it has the correct behaviour, I'll test it out today.